### ISSUE TITLE
Sincedb file won't be updated until getting all the logs

### ISSUE THREAD

**Issue by 0x4D31** [lukewaite/logstash-input-cloudwatch-logs#4]
Hi, Yesterday I restarted the logstash for some reasons, and after that it couldn't update the sincedb files anymore, because the log groups contains a huge amount of logs and they're growing every minute.

It seems the sincedb file won't be updated until getting all the logs in the specified Log Groups. So, when you want to read a large amount of logs with this plugin, if the logstash/plugin stops or restarts before getting all the logs, it will start getting the logs from the beginning (from the date/time that's written in sincedb file). And it takes a long time to reach the end or maybe it never reaches the end because of the growing number of logs, regular restarts and errors like this (execution expired):

```
{:timestamp=>"2016-01-19T02:03:30.537000+0000", :message=>"A plugin had an unrecoverable error. Will restart this plugin.\n  Plugin: <LogStash::Inputs::CloudWatch_Logs log_group=>\"sampleLogGroup\", region=>\"us-east-1\", sincedb_path=>\"/tmp/sincedb_something\", interval=>55, codec=><LogStash::Codecs::JSON charset=>\"UTF-8\">, debug=>false>\n  Error: execution expired", :level=>:error}
```

~almost similar issue in logstash-input-file plugin:
https://discuss.elastic.co/t/sincedb-doesnt-updated-until-tail-reaches-the-end-of-the-file/36116

**Comment by lukewaite** [comment]
This is definitely an issue if ingesting large volumes of older logs. I will work on a fix for this. PRs welcome.

**Comment by lukewaite** [comment]
I've just released `v1.0.0.pre`, a pre-release which resolves this issue. State is now persisted to the sincedb after each page of log events received from AWS' CWL api.

https://github.com/lukewaite/logstash-input-cloudwatch-logs/releases/tag/v1.0.0.pre
